libsolidity/experimental/ast/TypeSystem.h

```diff
@@ -74,6 +74,12 @@ class TypeEnvironment
 	TypeSystem& typeSystem() { return m_typeSystem; }
 	TypeSystem const& typeSystem() const { return m_typeSystem; }
 
+	bool isFixedTypeVar(Type const& _typeVar) const;
+	bool isGenericTypeVar(Type const& _typeVar) const;
+	bool hasGenericTypeVars(Type const& _type) const;
+	void fixTypeVars(std::vector<Type> const& _typeVars);
+	void generalizeTypeVars(std::vector<Type> const& _typeVars);
```

Comment:
This should never happen - you'd instead just freshen. It also *isn't* used currently, right? Let's keep it simple and make being fixed a monotonic property (i.e. you can fix, but not "un"fix things)

Reply:
Yeah, it's not used now but I expected it to be needed in the future. Otherwise there's no way to every unify anything with a fixed variable, even in a cloned environment.

Removed in any case.